CMakeLists.txt

```diff
@@ -513,6 +513,13 @@ endif()
 
 include(CMakePackageConfigHelpers)
 
+if(DEFINED OPENTELEMETRY_BUILD_DLL)
+  if(NOT MSVC)
+    message(WARNING "Build DLL is supposed to work with MSVC")
+  endif()
+  add_definitions(-DOPENTELEMETRY_BUILD_EXPORT_DLL)
```

Comment:
This has a global scope, for the entire built tree.

When OPENTELEMETRY_BUILD_DLL is set,
the content of the library (api, sdk, exporters, ext) should be build with _EXPORT_DDL,
but everything else that is not part of the library (examples, tests) should not see _EXPORT_DDL.

It seems this is the root cause for need of a _IMPORT_DDL flag.

Reply:
Yes, define `OPENTELEMETRY_BUILD_EXPORT_DLL` is global scope is intended than pushing it down into discrete libraries which should be compiled as DLL. So in cases of adding future component, it doesn't need to add this flag explicitly for DLL build (which could be error prone).